builder/builder.go

```diff
@@ -175,7 +175,7 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 		}
 		res := gwclient.NewResult()
 		if !b.builtMain {
-			ref, err := b.stateToRef(ctx, gwClient, b.targetPhaseState(mts.Final), mts.Final.Platform)
+			ref, err := b.stateToRef(ctx, gwClient, mts.Final.MainState, mts.Final.Platform)
```

Comment:
This change is a no-op - just made this change for clarity as it only applies to main state.